pantab/_reader.py

```diff
@@ -30,7 +23,12 @@ def _read_table(*, connection: tab_api.Connection, table: TableType) -> pd.DataF
     dtypes: Dict[str, str] = {}
     for column in columns:
         column_type = pantab_types._ColumnType(column.type, column.nullability)
-        dtypes[column.name.unescaped] = _tableau_to_pandas_type(column_type)
+        try:
+            dtypes[column.name.unescaped] = pantab_types._pandas_types[column_type]
+        except KeyError:
+            raise Exception(
```

Comment:
```suggestion
            raise TypeError(f"Column {column.name} has unsupported datatype {column.type}") from e
```

The raise...from syntax is usually more useful for developers since it gives the entire traceback. Additionally, you typically want to avoid throwing a base `Exception` in Python as its sometimes too generic to be useful. 

For example, if a user wanted to wrap this error on their end and just tried to catch `Exception`, that would cover this error as well as KeyboardInterrupts, amongst a variety of other things